cmd/cli-doc/cli-doc.go

```diff
@@ -86,11 +89,74 @@ func referencePrinter(command *cobra.Command, level int) string {
 		"```sh\n"+command.Example+"\n```",
 		command.Long,
 		tableOutput.String(),
-		"```sh\n"+GenerateCLIStructure()+"\n```",
+		"```sh\n"+fmt.Sprint(commandPrinter(command, 0))+"\n```",
 		commandOutput)
 }
 
+func getFlags(flags *pflag.FlagSet) []string {
+	var f []string
+	flags.VisitAll(func(flag *pflag.Flag) {
+		f = append(f, fmt.Sprintf("--%v", flag.Name))
+	})
+	return f
+}
+
+func flattenFlags(flags []string) string {
+	var flagString string
+	for _, flag := range flags {
+		flagString = flagString + flag + " "
+	}
+	return flagString
+}
+
+func commandPrinter(command *cobra.Command, level int) string {
+	var finalCommand string
+	// add indentation
+	for i := 0; i < level; i++ {
+		finalCommand = finalCommand + "    "
+	}
+	finalCommand = finalCommand +
+		command.Name() +
+		" " +
+		flattenFlags(getFlags(command.NonInheritedFlags())) +
+		": " +
+		command.Short +
+		"\n"
+	for _, subcommand := range command.Commands() {
+		finalCommand = finalCommand + commandPrinter(subcommand, level+1)
+	}
+	return finalCommand
+}
+
 // Generates and returns a markdown-formatted CLI reference page for Odo
-func GenerateCLIReference() string {
-	return referencePrinter(rootCmd, 0)
+func main() {
+	var clidoc = &cobra.Command{
```

Comment:
I might be missing something..
But why a cobra command for this?

Reply:
To be able to quickly support args since this command now replaces 2 commands and has to deal with args (which wasn't really the case before).